### DISCUSSION TITLE
Edge Functions sometimes have a 404 Error

### DISCUSSION THREAD

**Discussion by tranminhquang4421** [supabase/supabase#27485] [Questions]
Hi everyone, I have a question. 
Why do my edge functions frequently return a 404 error even though they were successfully invoked before?
Below is the response data of an invoked with a 404 error

```json
{
  "headers": [
    {
      "content_length": "88",
      "content_type": "application/json",
      "date": "Mon, 24 Jun 2024 07:10:00 GMT",
      "server": "cloudflare",
      "vary": "Accept-Encoding",
      "x_sb_edge_region": "ap-southeast-1",
      "x_served_by": "supabase-edge-runtime"
    }
  ],
  "status_code": 404
}
```

<img width="624" alt="Screenshot 2024-06-24 at 14 24 57" src="https://github.com/supabase/supabase/assets/142202992/fe336b47-5c8b-4f08-bcae-82c927c9eb3f">

**Comment by silentworks** [comment]
It's probably best to contact Supabase support about this https://supabase.com/dashboard/support/new

**Comment by kasrak** [comment]
Did you ever get this resolved? We're starting seeing a similar issue on one of our functions.

**Reply by tranminhquang4421** [reply]
@kasrak 
One reason is that I'm using GitHub Actions for auto-deploying edge functions. It's important to note that the GitHub runner might be executed in a region different from the one where the client is invoking the function. I'm not sure why Supabase selects the closest region based on the server location performing the deployment. This can lead to instability in the function across different regions, including the client's region.

To address this issue, I’ve deployed the functions in the same region as the client’s invocation by setting the x-region header accordingly

**Reply by kasrak** [reply]
Interesting, thank you!

We figured out our issue: we had a package in our deno import map that was missing a version ("npm:braintrust" instead of "npm:braintrust@0.0.133"). Once we added the version number, the function started working.
